src/linker/Linker/LinkerAttributesInformation.cs

```diff
@@ -12,9 +13,11 @@ namespace Mono.Linker
 {
 	readonly struct LinkerAttributesInformation
 	{
-		readonly Dictionary<Type, List<Attribute>>? _linkerAttributes;
+		readonly List<Tuple<Type, List<Attribute>>>? _linkerAttributes;
```

Comment:
Nit: I find it a bit easier to read to use the named values in this case, so for example:

```suggestion
		readonly List<(Type Type, List<Attribute> Attributes)>? _linkerAttributes;
```

That way when you're referencing the pieces you can use descriptive names. So instead of `x.Item1` it would read `x.Type` below.

It does make the type a bit longer though, but personally I still prefer the descriptive names.

Note that you only need to specify the names in the declaration, no need to do so in the "new" call (at least I think).